raiden/smart_contracts/ChannelManagerContract.sol

```diff
@@ -40,10 +41,19 @@ contract ChannelManagerContract {
         address[] memory result;
         NettingChannelContract channel;
 
-        result = new address[](data.all_channels.length * 2);
+        uint open_channels_num = 0;
+        for (i = 0; i < data.all_channels.length; i++) {
+            if (contractExists(data.all_channels[i])) {
+                open_channels_num += 1;
+            }
+        }
+        result = new address[](open_channels_num * 2);
 
         pos = 0;
         for (i = 0; i < data.all_channels.length; i++) {
+            if (!contractExists(data.all_channels[i])) {
```

Comment:
This is a constant function to be called by the client only with a `call` and not a transaction, thus it spends no gas.